napari/_qt/qt_mode_buttons.py

```diff
@@ -0,0 +1,35 @@
+from qtpy.QtWidgets import QRadioButton, QPushButton
+
+
+class QtModeRadioButton(QRadioButton):
+    def __init__(
+        self, layer, button_name, mode=None, tool_tip=None, checked=False
```

Comment:
Looking at this again I guess `mode=None` doesn't really make sense here as this button would never do anything without mode passed. Maybe ok to make that not keyword arg, but I still might be the `*` before `tool_tip` for all these so `tool_tip` had to be keyword arg, and then go back and add `tool_tip=` to all the places where the function is called, it might make them more readable. If you prefer otherwise @tlambert03 that's totally fine, but curious why?

Reply:
yeah, the mode arg is pretty important here.  I can make it required again if you want.  I don't think it matters too much since it's only going to be used by GUI builders.  My inclination is to make it required.  
I don't care too much either way regarding the tool_tip.  I haven't personally used the `*` enforcement much in my own stuff, but I don't feel strongly about it.  I guess we're just saying we never want to see this used without someone labeling it as a tooltip for readability?

Reply:
making mode required sounds good.
yeah on `tool_tip` that matches how I think about it, @jni has good intuition for this stuff, maybe he  can weigh in, but I think he's tended to be in favor of at least using `tool_tip=` like stuff in our internal code instead of position stuff even if its not required for readability, so we could just do that without the `*`

Reply:
> I guess we're just saying we never want to see this used without someone labeling it as a tooltip for readability?

Yes, exactly. For example, all of the QWidget(parent, 0, 1, 1, 2, 75, whatever) calls drive me up the wall. Where arguments are not essentially self-documenting (e.g. out = gaussian_filter(arr)), keyword-only arguments should be used.

Also imho, `tooltip` over `tool_tip`. The joined form is a word now as far as I'm concerned.

Reply:
sounds good, I can change that.  also, just cause I don't want you to think i'm hastily adding new classes without thinking about them 😂, that class was just transplanted from the qt_shapes layer in master (but I agree this is the time to reconsider it and I could/should have done that):
https://github.com/napari/napari/blob/0f2f8ef98b5f22983283cbd6072f92b267304fbe/napari/_qt/layers/qt_shapes_layer.py#L239-L241